dotCMS/src/main/java/com/dotmarketing/db/DbConnectionFactory.java

```diff
@@ -269,6 +281,14 @@ public static boolean connectionExists() {
         return isCreated;
     } // connectionExists.
 
+    public static Optional<Throwable> getCreatorStack() {
+        if (connectionExists()) {
```

Comment:
my preference is one line return return connectionExists()?  Optional.ofNullable(creatorStack.get()):  Optional.empty();